src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5571,7 +5638,7 @@ private BoundExpression BindUnexpectedComplexElementInitializer(InitializerExpre
         }
 
         private BoundExpression BindCollectionInitializerElementAddMethod(
-            ExpressionSyntax elementInitializer,
+            SyntaxNode elementInitializer,
```

Comment:
When is this not an expression syntax? Or was this just changed for convenience (removing need for casting?)

Reply:
Currently, we can end up here for spread elements and dictionary elements, when the result was a `BoundBadExpression`. But it seems questionable to call this method in those cases. I've added a PROTOTYPE comment to revisit in an upcoming PR.